### PR TITLE
Edit Library and Show's Advanced Settings

### PR DESCRIPTION
**Pull request by blacktwin**
Allow for editing a show's advanced settings

`show.editAdvanced(showOrdering='aired', autoDeletionItemPolicyWatchedLibrary=0)`

Allow for editing a library's advanced settings

`library.editAdvanced(enableBIFGeneration=0, collectionMode=2)`

### CODE REVIEW

**Comment by blacktwin** [comment]
I can move these methods to `settings.Preferences` if I carry over the show's key into `settings.Preferences`'s `_initpath`.

```python
    def preferences(self):
        """ Returns a list of :class:`~plexapi.settings.Preferences` objects. """
        items = []
        data = self._server.query(self._details_key)
        for item in data.iter('Preferences'):
            for elem in item:
                setting = settings.Preferences(data=elem, server=self._server)
                setting._initpath = self.key
                items.append(setting)

        return items

```

This way the settings could be updated from the show object in mass or from the preference object.

**Comment by blacktwin** [comment]
I should add some tests to this. Switch settings, reload, assert.

**Comment by blacktwin** [comment]
@Hellowlol this should be ready for review.

File: plexapi/video.py [hunk 1]
```diff
@@ -442,10 +442,38 @@ def preferences(self):
...
+    def editAdvanced(self, **kwargs):
+        """ Edit a show's advanced settings. """
+        data = {}
+        key = '%s/prefs?' % self.key
+        preferences = {pref.id: list(pref.enumValues.keys()) for pref in self.preferences()}
+        for settingID, value in kwargs.items():
+            try:
+                enumValues = [int(x) for x in preferences.get(settingID)]
+            except ValueError:
+                enumValues = [x.decode() for x in preferences.get(settingID)]
```
[the remark this answers is not in the document]
Comment (blacktwin) [reply plexapi/video.py:460]: The `.value` can be classed as a `byte` that is either an `int` or `str`. Majority of the time their `int` `-1, 1, 2` something like that.

File: plexapi/settings.py [hunk 2]
```diff
@@ -167,3 +167,12 @@ class Preferences(Setting):
...
     TAG = 'Preferences'
     FILTER = 'preferences'
+
+    def _default(self):
+        """ Set the default value for this setting."""
+        key = '%s/prefs?' % self._initpath
+        if self.type == 'int':
+            url = key + '%s=%s' % (self.id, self.default)
+        else:
+            url = key + '%s=%s' % (self.id, self.default.decode())
```
[the remark this answers is not in the document]
Comment (blacktwin) [reply plexapi/settings.py:177]: `.default` is a value so it also is a `byte` that is either an `int` or `str`.

File: plexapi/video.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (blacktwin) [reply plexapi/video.py:460]:
```python
    _bool_cast = lambda x: True if x == 'true' or x == '1' else False
    _bool_str = lambda x: str(x).lower()
    _str = lambda x: str(x).encode('utf-8') # py2 dependant?
    TYPES = {
        'bool': {'type': bool, 'cast': _bool_cast, 'tostr': _bool_str},
        'double': {'type': float, 'cast': float, 'tostr': _str},
        'int': {'type': int, 'cast': int, 'tostr': _str},
        'text': {'type': str, 'cast': _str , 'tostr': _str},
    }
```

Updating the the `tostr: _str` references to `tostr: str` should resolve the issue.

**Comment by jjlawren** [comment]
Calling `PlexServer.settings.get("FriendlyName")` is returning an empty value for some reason in [this test](https://github.com/pkkid/python-plexapi/blob/8410d81520bfe6e9ba20718953baf14eecc18c6a/tests/test_settings.py#L5-L11), and only for this PR. Could one of the changes/tests be inadvertently clearing it somehow? Or maybe it just needs a `reload()` at the start of the failing test?

**Comment by blacktwin** [comment]
This PR should be gtg.

**Pull request merged**